stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -0,0 +1,634 @@
+module Artifacts
+
+import Base: get, SHA1
+using Base.BinaryPlatforms, Base.TOML
+
+export artifact_exists, artifact_path, remove_artifact, verify_artifact,
+       artifact_meta, artifact_hash, find_artifacts_toml, @artifact_str
+
+"""
+    parse_toml(path::String)
+
+Uses Base.TOML to parse a TOML file
+"""
+function parse_toml(path::String)
+    p = Base.TOML.Parser()
+    Base.TOML.reinit!(p, read(path, String); filepath=path)
+    return Base.TOML.parse(p)
+end
+
+# keep in sync with Base.project_names and Base.manifest_names
+const artifact_names = ("JuliaArtifacts.toml", "Artifacts.toml")
+
+const ARTIFACTS_DIR_OVERRIDE = Ref{Union{String,Nothing}}(nothing)
+"""
+    with_artifacts_directory(f::Function, artifacts_dir::String)
+
+Helper function to allow temporarily changing the artifact installation and search
+directory.  When this is set, no other directory will be searched for artifacts, and new
+artifacts will be installed within this directory.  Similarly, removing an artifact will
+only effect the given artifact directory.  To layer artifact installation locations, use
+the typical Julia depot path mechanism.
+"""
+function with_artifacts_directory(f::Function, artifacts_dir::String)
+    try
+        ARTIFACTS_DIR_OVERRIDE[] = artifacts_dir
+        f()
```

Comment:
```suggestion
        string(f())::String
```

maybe

Reply:
Why is `f()` returning a `String`?  I don't think that's true.

Reply:
I think I mixed this up with some other function...